### PR TITLE
Update boolean.py library

### PR DESCRIPTION
**Pull request by majurg**
* Bump boolean.py library to v3.7, which adds the ability to sort OR
  leave boolean expressions unsorted.
* Add test case for simplify() sorting

Addresses: #32

Signed-off-by: Steven Esser <sesser@nexb.com>

### CODE REVIEW

File: tests/test_license_expression.py [hunk 1]
```diff
@@ -552,6 +552,14 @@ def test_parse_simplify_and_contain_and_equal(self):
         expr3 = licensing.parse('mit and LGPL2.1')
         self.assertTrue(expr3 in expr2)
 
+    def test_parse_simplify_no_sort(self):
+        licensing = Licensing()
+
+        expr = licensing.parse('gpl-2.0 OR apache-2.0 OR gpl-2.0')
+
+        self.assertEqual(expr.simplify(), expr.simplify(sort=False))
```
Comment (pombredanne) [inline tests/test_license_expression.py]: with `gpl-2.0 OR apache-2.0 OR gpl-2.0` I would expect the simplification to be `gpl-2.0 OR apache-2.0` ... would you?

Also a small nitpicking: may be we do not need so many empty lines?

Comment (majurg) [reply tests/test_license_expression.py]: This is checking to make sure the expressions themselves are equal, even if the "sort" of the expression is different. For instance `mit OR gpl-2.0 === gpl-2.0 OR mit` are technically the same data-wise and interpretation-wise. 

The only difference between the two expressions is the ordering, which the following test on line 560 handles.

Comment (majurg) [reply tests/test_license_expression.py]: @pombredanne and we probably do not need so many lines, I will fix this ;)

**Comment by majurg** [comment]
@pombredanne ping

the travis builds are timing-out for some unknown reason.

**Comment by pombredanne** [comment]
I have restarted the builds

**Comment by majurg** [comment]
there might be something wrong with the following lines in travis on osx that are causing it to hang? Do we need to use pyenv for these?:
```
if [[ $TRAVIS_OS_NAME == 'osx' ]]; then
        rm -rf ~/.pyenv
        git clone https://github.com/pyenv/pyenv.git ~/.pyenv
        echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bash_profile
        echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bash_profile
        echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init -)"\nfi' >> ~/.bash_profile
        exec "$SHELL"
    fi
```

**Comment by pombredanne** [comment]
>Do we need to use pyenv for these?

No. Anything works and if you look at the latest travis from scancode, this is quite different now
https://github.com/nexB/scancode-toolkit/blob/develop/.travis.yml

**Comment by majurg** [comment]
@pombredanne Everything is working now.

**Comment by pombredanne** [comment]
Thank you! LGTM... merging.

**Comment by pombredanne** [comment]
@majurg what is you Pypi username? I will grant you access to make a release

**Pull request merged**